src/utilities/metrics.js

```diff
@@ -1,24 +1,64 @@
-import Amplitude from 'amplitude-js'
+/**
+ * @prettier
+ */
+import throttle from "lodash/throttle";
 
-const instance = Amplitude.getInstance();
-instance.init(process.env.AMPLITUDE_KEY)
+const METRICS_ENDPOINT = "https://api.amplitude.com/2/httpapi";
+let cache = [];
 
-export function logEvent(type, properties) {
-  // In development, track when events are fired for debugging purposes.
-  if (process.env.NODE_ENV === 'development') {
-    console.log(`[METRICS]: "${type}", ${JSON.stringify(properties)}`)
+const uploadMetrics = throttle(() => {
+  const toUpload = cache;
+  cache = [];
+  if (!process.env.AMPLITUDE_KEY) {
+    // eslint-disable-next-line no-console
+    console.log("Not uploading metrics", toUpload);
+    return;
+  }
+  fetch(METRICS_ENDPOINT, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      api_key: process.env.AMPLITUDE_KEY,
+      events: toUpload,
+    }),
+  });
+}, 500);
+
+const getUserId = () => {
+  const storedId = localStorage.getItem("metrics_user_id");
+  if (!storedId) {
+    // Create a random ID by taking the decimal portion of a random number
+    const newId = Math.random()
+      .toString()
+      .split(".")[1];
+    localStorage.setItem("metrics_user_id", newId);
+    return newId;
```

Comment:
if !storeId, do you need to set `storedId` to `newId` in this conditional so you have a value to return at the bottom of this?

Reply:
Nah this is an early return, no value needed in `storedId`. 
```suggestion
    storedId = newId;
```
would also work but then I'd also have to do `let storedId…` so 🤷

Reply:
ah you're right